### DISCUSSION TITLE
upscale model parameter

### DISCUSSION THREAD

**Discussion by kalle07** [lllyasviel/Fooocus#3211] [Q&A] [answered]
maybe a stupid question

ok here i can upscale, but no parameter ? denoise , steps ?
![grafik](https://github.com/lllyasviel/Fooocus/assets/118767589/da8f1dfb-b822-4c26-98f4-102bb81cd723)

and if, where i can change the upscale-model, is "bin" and "pth" possible ?

**Comment by mashb1t** [comment, marked as answer]
for all upscale modes except fast 2x, the image is re-generated with a fixed low denoise strength (vary is more).
https://github.com/lllyasviel/Fooocus/blob/main/modules/async_worker.py#L633 / https://github.com/lllyasviel/Fooocus/blob/main/modules/async_worker.py#L554

You can override this in the developer debug mode settings.

Upscale is done using an upscale model, which can be changed here (+ config.py for automated download): https://github.com/lllyasviel/Fooocus/blob/main/modules/upscaler.py#L10

**Reply by kalle07** [reply]
thx so long...
only bin files work? ok they are rare often pth is aviable ... it is easy to compile? or you have a source of some bin upscalers

**Reply by mashb1t** [reply]
Probably also .pt/.pth and every filetype supported by `torch.load` (https://pytorch.org/docs/2.1/generated/torch.load.html) are supported, but you might have to adjust the logic of the upscaler as all items have to have RDB as basis for ESRGAN to work on the model. If you want to upscale without ESRGAN you'll have to manually adjust the code further.

**Reply by kalle07** [reply]
oh iam not a coder ...
a bit pity that there no log in CMD or in the image which upscaler is loaded or used ... i could make a request
